app/Helpers/database/forum.php

```diff
@@ -335,23 +295,24 @@ function getTopicCommentCommentOffset(int $forumTopicID, int $commentID, int $co
     return false;
 }
 
-function generateGameForumTopic(string $user, int $gameID, ?int &$forumTopicID): bool
+function generateGameForumTopic(User $user, int $gameId): ?ForumTopicComment
 {
-    if ($gameID == 0) {
-        return false;
+    if ($gameId === 0) {
+        return null;
     }
 
-    $game = Game::with('system')->find($gameID);
+    $game = Game::with('system')->find($gameId);
     if (!$game) {
-        return false;
+        return null;
     }
 
-    if ($game->ForumTopicID > 0 && ForumTopic::where('ID', $game->ForumTopicID)->exists()) {
-        // valid topic already exists
-        return false;
+    // If a valid forum topic already exists for the game, bail.
+    if ($game->ForumTopicID > 0 && ForumTopic::find($game->ForumTopicID)->exists()) {
+        return null;
     }
 
-    $forumID = match ($game->system->ID) {
+    // TODO we probably can't get away with hardcoding this indefinitely.
```

Comment:
This is broken in more ways than just being hard-coded. Everything that's not in the list goes straight to the Mega Drive forum.

Once we have a system editor, that can be used to associate/create forums for each system. See #650.

There's also plans to allow each game to have it's own forum (instead of just an official forum topic). See #661. At that point, it may be moot to have system-level forums.